src/SamplesApp/SamplesApp.UITests.Generator/SnapShotTestGenerator.cs

```diff
@@ -215,45 +220,16 @@ private static object Sanitize(string category)
 
 		private static Compilation GetCompilation(GeneratorExecutionContext context)
 		{
-			// Used to get the reference assemblies
-			var devEnvDir = context.GetMSBuildPropertyValue("MSBuildExtensionsPath");
-
-			if (devEnvDir.StartsWith("*"))
-			{
-				throw new Exception($"The reference assemblies path is not defined");
-			}
-
-			var ws = new AdhocWorkspace();
-
-			var project = ws.CurrentSolution.AddProject("temp", "temp", LanguageNames.CSharp);
-
-			var referenceFiles = new[] {
-				typeof(object).Assembly.CodeBase,
-				typeof(Attribute).Assembly.CodeBase,
-			};
-
-			foreach (var file in referenceFiles.Distinct())
-			{
-				project = project.AddMetadataReference(MetadataReference.CreateFromFile(new Uri(file).LocalPath));
-			}
-
-			project = AddFiles(context, project, "UITests.Shared");
-			project = AddFiles(context, project, "SamplesApp.UnitTests.Shared");
-
-			var compilation = project.GetCompilationAsync().Result;
-
-			return compilation;
+			return context.Compilation.AddSyntaxTrees(GetSyntaxTrees(context, "UITests.Shared").Concat(GetSyntaxTrees(context, "SamplesApp.UnitTests.Shared")));
```

Comment:
Actually, this does not seem to work properly. Take a look at the snapshot tests on iOS, they don't run the tests, likely showing that the files are not picked up properly.

Reply:
Umm... That's strange

![image](https://user-images.githubusercontent.com/31348972/192275799-5ed80dcf-4aa4-4587-8217-6710cb224832.png)

It seems to be working on my side.

![image](https://user-images.githubusercontent.com/31348972/192275960-3589570b-fd6b-4ba3-b299-c53872c7d493.png)